### PR TITLE
Only generate output files with content

### PR DESCRIPTION
**Pull request by johnlister**
- [ ] I've added tests (if it's a bug, feature or enhancement)
- [ ] I've adjusted the documentation (if it's a feature or enhancement)
- [ ] The test suite passes (run `bundle exec rspec` to verify this)

## Summary
Fixes https://github.com/boltops-tools/terraspace/issues/284

## Version Changes

patch

### CODE REVIEW

File: lib/terraspace/compiler/writer.rb [hunk 1]
```diff
@@ -34,7 +34,7 @@ def write(content)
       if content.respond_to?(:path) # IO filehandle
         FileUtils.cp(content.path, dest_path) # preserves permission
       else # just content
-        IO.write(dest_path, content, mode: "wb")
+        IO.write(dest_path, content, mode: "wb") unless content.nil?
       end
       logger.debug "Created #{Terraspace::Util.pretty_path(dest_path)}"
```
Comment (tongueroo) [inline lib/terraspace/compiler/writer.rb:39]: Probably add the same `if content.nil?` to prevent the `logger.debug` also. Thanks!

**Comment by tongueroo** [comment]
Thanks for the PR! 😄

Ran the terraspace acceptance tests pipeline on it. Annoyingly, this slight change breaks `terraspace all` 🤦🏻‍♂️

Pipeline with boltops-tools master (passes)

terraspace pipeline good master

Pipeline with johnlister fork (fails)

terraspace pipeline broken pr

Took a look at the `terraspace all` codebuild logs and sadly it didn't help much. Will have to try to improve the TSC logging when get a chance. 

terraspace codebuild terraspace all logging

Will have to dig into this more when get a chance. Here's the build script https://github.com/boltops-tools/terraspace/blob/master/.cody/all/bin/build.sh The particularly test checks that https://github.com/boltops-tools/terraspace-graph-demo still works.  Will have to just clone it down locally and test to investigate further.

**Comment by johnlister** [comment]
I'll have a look, to be honest didn't run terraspace all locally after applying it. Suspect something relies on an empty file?

**Comment by tongueroo** [comment]
> RE: Suspect something relies on an empty file?

Guessing the same thing.

**Comment by johnlister** [comment]
Odd, just ran it locally on my branch and it succeeded - only change I made to the demo graph was to use a local backend, applied the change to your latest master branch and it failed, but re-running a second time passed.  Couldn't see any empty files in the demo... Will re-look later

**Comment by tongueroo** [comment]
Bummer 😒 Thanks for the update.

**Comment by tongueroo** [comment]
Ran the pipeline again and it passed this time. It may be something intermittent 🤦🏻‍♂️  Merging as-is.

**Comment by tongueroo** [comment]
Released in 2.2.3

**Pull request merged**